pkg/controllers/workgenerator/controller_test.go

```diff
@@ -2908,6 +3079,94 @@ func TestUpdateBindingStatusWithRetry(t *testing.T) {
 	}
 }
 
+func TestSyncApplyStrategy(t *testing.T) {
+	bindingName := "test-binding-1"
+	workName := "test-work-1"
+
+	testCases := []struct {
+		name              string
+		resourceBinding   *fleetv1beta1.ClusterResourceBinding
+		work              *fleetv1beta1.Work
+		wantUpdated       bool
+		wantApplyStrategy *fleetv1beta1.ApplyStrategy
+	}{
+		{
```

Comment:
add a diff test case? the work has something but the binding has something else?

Reply:
Added 👌